sources/packages/backend/apps/queue-consumers/src/services/workflow/workflow-enqueuer.service.ts

```diff
@@ -115,4 +159,56 @@ export class WorkflowEnqueuerService {
     }
     return summary;
   }
+
+  /**
+   * Queue the next pending assessment for cancellation for an application.
+   * @param application application with pending assessments.
+   * @returns process summary.
+   */
+  private async queueAssessmentCancellation(
+    application: Application,
+  ): Promise<ProcessSummary> {
+    const summary = new ProcessSummary();
+    try {
+      summary.info(
+        `Queueing next pending assessment for application id ${application.id}.`,
+      );
+      const [nextAssessment] = application.studentAssessments;
+      summary.info(
+        `Found ${application.studentAssessments.length} pending assessment(s) cancellation. Queueing assessment ${nextAssessment.id} for cancellation.`,
+      );
+      summary.info(
+        `Updating assessment status to ${StudentAssessmentStatus.CancellationQueued}.`,
+      );
+      const now = new Date();
+      const systemUser = await this.systemUsersService.systemUser();
+      const assessmentUpdateResults = await this.studentAssessmentRepo.update(
```

Comment:
Not a part of the PR. It will be nice to update audit fields on other method `queueNextAssessment` as well